homeassistant/components/samsungtv/config_flow.py

```diff
@@ -0,0 +1,198 @@
+            )
+
+            if _is_already_configured(self.hass, ip_address):
+                return self.async_abort(reason="already_configured")
+
+            self._host = user_input[CONF_HOST]
+            self._ip = self.context[CONF_IP_ADDRESS] = ip_address
+            self._title = user_input[CONF_NAME]
+
+            result = await self.hass.async_add_executor_job(
+                _try_connect, self._host, self._title
+            )
+
+            if result != RESULT_SUCCESS:
+                return self.async_abort(reason=result)
+            return self._get_entry()
+
+        return self.async_show_form(step_id="user", data_schema=DATA_SCHEMA)
+
+    async def async_step_ssdp(self, user_input=None):
+        """Handle a flow initialized by discovery."""
+        ip_address = await self.hass.async_add_executor_job(
+            _get_ip, user_input[ATTR_HOST]
+        )
+
+        if any(
+            ip_address == flow["context"].get(CONF_IP_ADDRESS)
+            for flow in self._async_in_progress()
+        ):
+            return self.async_abort(reason="already_in_progress")
+
+        if _is_already_configured(self.hass, ip_address):
+            return self.async_abort(reason="already_configured")
+
+        self._host = user_input[ATTR_HOST]
+        self._ip = self.context[CONF_IP_ADDRESS] = ip_address
+        self._manufacturer = user_input[ATTR_MANUFACTURER]
+        self._model = user_input[ATTR_MODEL_NAME]
+        self._name = user_input[ATTR_NAME]
+        if self._name.startswith("[TV]"):
+            self._name = self._name[4:]
+        self._title = f"{self._name} ({self._model})"
+        self._uuid = user_input[ATTR_UDN]
+        if self._uuid.startswith("uuid:"):
+            self._uuid = self._uuid[5:]
+
+        return await self.async_step_confirm()
+
+    async def async_step_confirm(self, user_input=None):
+        """Handle user-confirmation of discovered node."""
+        if user_input is not None:
+            result = await self.hass.async_add_executor_job(
+                _try_connect, self._host, self._name
+            )
+
+            if result != RESULT_SUCCESS:
+                return self.async_abort(reason=result)
+            return self._get_entry()
+
+        return self.async_show_form(
+            step_id="confirm", description_placeholders={"model": self._model}
+        )
+
+    async def async_step_import(self, user_input):
+        """Handle a flow initialized by config import."""
+        ip_address = await self.hass.async_add_executor_job(
+            _get_ip, user_input.get(CONF_HOST)
+        )
+
+        self._host = user_input.get(CONF_HOST)
+        self._ip = self.context[CONF_IP_ADDRESS] = ip_address
+        self._mac = user_input.get(CONF_MAC)
+        self._port = user_input.get(CONF_PORT)
+        self._timeout = user_input.get(CONF_TIMEOUT)
+        self._title = user_input.get(CONF_NAME)
+
+        return self._get_entry()
```

Comment:
We need to check the connection before creating the entry here too. Can we pass the config to the user step?

Reply:
Sure, but why do we need to check? The TV could be of when starting HA.

Reply:
The import should only create flow and config entry if there's no corresponding entry already. So I think it should be ok to require the TV to be on during the first setup/configuration.

Reply:
Ah yes, wasn't aware the import needs to run only once, not on every start.

Reply:
Ah yes, wasn't aware the import needs to run only once, not on every start.

Reply:
As long as the config yaml doesn't change, we don't need to create a new entry since it will be loaded from storage.